ow_dynamic_terrain/src/DynamicTerrainVisual.cpp

```diff
@@ -13,129 +13,133 @@ using namespace Ogre;
 
 namespace ow_dynamic_terrain
 {
-class DynamicTerrainVisual : public VisualPlugin, public DynamicTerrainBase
+
+GZ_REGISTER_VISUAL_PLUGIN(DynamicTerrainVisual)
+
+DynamicTerrainVisual::DynamicTerrainVisual() :
+  DynamicTerrainBase{ "ow_dynamic_terrain", "DynamicTerrainVisual" }
 {
-public:
-  DynamicTerrainVisual() : DynamicTerrainBase{ "ow_dynamic_terrain", "DynamicTerrainVisual" }
-  {
-  }
+}
 
-  void Load(VisualPtr /*visual*/, sdf::ElementPtr /*sdf*/) override
-  {
-    Initialize("visual");
-
-    m_on_prerender_connection = gazebo::event::Events::ConnectPreRender([this]() {
-      // Initialize terrain with Ogre-generated normal map.
-      applyNormalMap();
-      // Disconnect so this method will not be called again.
-      m_on_prerender_connection.reset();
-    });
-  }
+void DynamicTerrainVisual::Load(VisualPtr /*visual*/, sdf::ElementPtr /*sdf*/)
+{
+  Initialize("visual");
 
-private:
-  static inline float getHeightInWorldCoords(const Ogre::Terrain* terrain, int x, int y)
-  {
-    auto value = terrain->getHeightAtPoint(x, y);
-    value += terrain->getPosition().z;
-    return value;
-  }
+  m_on_prerender_connection = gazebo::event::Events::ConnectPreRender([this]() {
+    initTextureUnits();
+  });
+}
 
-  static inline void setHeightFromWorldCoords(Ogre::Terrain* terrain, int x, int y, float value)
-  {
-    value -= terrain->getPosition().z;
-    terrain->setHeightAtPoint(x, y, value);
-  }
+float DynamicTerrainVisual::getHeightInWorldCoords(const Ogre::Terrain* terrain, int x, int y)
+{
+  auto value = terrain->getHeightAtPoint(x, y);
```

Comment:
Is it not valid or safe to declare `value` a float?

Reply:
This was existing code, but it got split into .h and .cpp files. I agree this should be float, but I would prefer to do this sort of cleanup in a separate ticket.

Reply:
I have a mixture of opinions about `auto`, but it does seem unnecessary here unless an `auto` return type is used, but then that would make the function's prototype far less descriptive... 

My feelings are `auto` usage should be very case-by-case. Sometimes I don't care about the type, other times I want to be very explicit about it if for no other reason than the code self-documents better.

This is just discussion by the way, not a change request. I'm fine with keeping the definition as it was.